contrib/uriparser/CMakeLists.txt

```diff
@@ -62,7 +62,7 @@ add_sapi_library(
     uriFreeUriMembersA
 
   INPUTS
-    "${CMAKE_BINARY_DIR}/uri.gen.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/uri.gen.h"
```

Comment:
Shouldn't this be PROJECT_BINARY_DIR?

Reply:
Are there any cases where these differ?

Reply:
`CMAKE_BINARY_DIR` refers to the top level binary directory, which may be the one used by the embedding project itself.\
`CMAKE_CURRENT_BINARY_DIR` refers to the binary directory that CMake is currently processing, so the one for contrib/uriparser in this case.\
Finally, `PROJECT_BINARY_DIR` refers to the binary directory of the most recent call to `project()`. This is the most "stable", as it will not change depending on how the uriparse `CMakeListst.txt` is consumed.